proto/redpanda/api/dataplane/v1alpha2/pipeline.proto

```diff
@@ -0,0 +1,386 @@
+      response_body: "pipeline"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Starts a stopped Redpanda Connect pipeline."
+      responses: {
+        key: "200"
+        value: {
+          description: "Started"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1alpha2.Pipeline"}
+          }
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Redpanda Connect pipeline not found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+  }
+}
+
+message PipelineCreate {
+  string display_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+    (buf.validate.field).string.min_len = 3,
+    (buf.validate.field).string.max_len = 128
+  ];
+  string description = 2 [(buf.validate.field).string.max_len = 256];
+
+  string config_yaml = 3 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  Pipeline.Resources request = 4 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  Pipeline.Resources limit = 5;
+}
+
+message CreatePipelineRequest {
+  PipelineCreate pipeline = 1;
+}
+
+message CreatePipelineResponse {
+  Pipeline pipeline = 1;
+}
+
+message GetPipelineRequest {
+  string id = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_/]+$"
+  ];
+}
+
+message GetPipelineResponse {
+  Pipeline pipeline = 1;
+}
+
+message ListPipelinesRequest {
+  Filter filter = 1;
+  int32 page_size = 2 [
+    (buf.validate.field).int32 = {
+      gte: -1,
+      lte: 1000
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Limit the paginated response to a number of items. Defaults to 100. Use -1 to disable pagination.",
+      minimum: -1,
+      maximum: 1000
+    }
+  ];
+  string page_token = 3;
+
+  message Filter {
+    string display_name = 1;
+  }
+}
+
+message ListPipelinesResponse {
+  repeated Pipeline pipelines = 1;
+  string next_page_token = 2;
+}
+
+message PipelineUpdate {
+  string display_name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+    (buf.validate.field).string.min_len = 3,
+    (buf.validate.field).string.max_len = 128
+  ];
+  string description = 2 [(buf.validate.field).string.max_len = 256];
+
+  string config_yaml = 3 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  Pipeline.Resources request = 4 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
```

Comment:
Do we want to remove this required too?

Reply:
Good catch. In updates everything should be optional.